workers/unity/Packages/com.improbable.gdk.testutils/TestLogDispatcher.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+using Improbable.Worker.Core;
+using NUnit.Framework;
+using UnityEngine;
+
+namespace Improbable.Gdk.TestUtils
+{
+    /// <summary>
+    ///     A ILogDispatcher implementation designed to be used in testing. This replaces the LogAssert approach with
+    ///     a more specialised one.
+    ///
+    ///     The expected usage is to use EnterExpectingScope() with a using block. This methods returns a Disposable
+    ///     object which you can mark logs as expected. When the object is disposed - it will assert against any logs.
+    /// </summary>
+    public class TestLogDispatcher : ILogDispatcher
```

Comment:
This is an awful lot of new code that's not actually used anywhere in this PR...

Reply:
You missed the whole thing with this - this is used in a couple of the tests as well :)